File: app/src/main/java/com/paragon/sensonic/ui/activities/login/LoginNavigator.java
package com.paragon.sensonic.ui.activities.login;

import com.paragon.sensonic.auth.dto.PasswordLessLogin;

public interface LoginNavigator {
    void init();
    void onLoginClick();
    void onCountyCodeClick();
    void onMobileClick();
    void onEmailClick();
    void onShowProgress();
    void onHideProgress();
    void setErrorText(boolean b, String errorText);
    void onSuccess(PasswordLessLogin response);
    void onError(String error);
}

File: app/src/main/java/com/paragon/sensonic/ui/activities/login/LoginNavigator.kt
package com.paragon.sensonic.ui.activities.login

import com.paragon.sensonic.auth.dto.PasswordLessLogin

interface LoginNavigator {
    fun init()
    fun onLoginClick()
    fun onCountyCodeClick()
    fun onMobileClick()
    fun onEmailClick()
    fun onShowProgress()
    fun onHideProgress()
    fun setErrorText(b: Boolean, errorText: String)
    fun onSuccess(response: PasswordLessLogin)
    fun onError(error: String)
}